### ISSUE TITLE
Separate the Auth package into it's own library.

### ISSUE THREAD

**Issue by gaurangrshah** [supabase/supabase#866]
# Feature request

I am currently implementing authentication using supabase within a nextjs application. Although the setup is straight forward and works pretty well, I'm struggling with the nuances. 

For example currently when a user logs out they are still able to get a valid session from the server, because there is still a valid session produced when fetching a session with `supabase.auth.api.getUser(token)`.

Although the user is logged out on the client side, my serverSideProps can still execute this request while the session is still valid.  

## Describe the solution you'd like

Simply put I'd really like to be able to use supabase's `Auth` without having to take on the overhead of the entire `@supabase/ui` library. 

## Describe alternatives you've considered

I'm currently handling this by also verifying a valid session on the client side and redirecting the user if the client-side session is no longer valid.

**Comment by kiwicopple** [comment]
Hi @gaurangrshah 

These are already separate libraries:

- `supabase-js`: https://github.com/supabase/supabase-js
- auth (`gotrue-js`): https://github.com/supabase/gotrue-js  (supabase-js uses this)
- supabase [UI](https://github.com/supabase/ui): you don't actually need to use the UI library, you can just use `supabase-js` or `gotrue-js`


> Simply put I'd really like to be able to use supabase's Auth without having to take on the overhead of the entire @supabase/ui library.

It's definitely possible, just using the supabase-js library: https://supabase.io/docs/client/auth-signup

I'll close this for now, but if I misunderstood the problem, feel free to re-open

**Comment by gaurangrshah** [comment]
I guess I was not aware of this. Good to know. Thank you. So maybe if this is the case, then I guess my issue is far simpler, and maybe I'm just doing something wrong -- or more likely not doing it the right way when it comes to logging users out. 

Let me ask you this, how do I go about invalidating the cookie token from supabase when a user signs out on the client-side? How do you recommend handling the fact that the token will still persist and be valid? Or should I just not even be concerned with this, and allow the user to be signed back in automatically if the token is still valid even though the logged out? 

Sorry if this seems like a "naive" question, I am def. confused about this. And @kiwicopple thank you for all of your help thus far. It has been greatly appreciated!

**Comment by kiwicopple** [comment]
This one is a good example of how to use cookies on the server side: https://github.com/supabase/supabase/tree/master/examples/nextjs-with-supabase-auth

> how do I go about invalidating the cookie token from supabase when a user signs out on the client-side?

Our Auth library already does this [here](https://github.com/supabase/gotrue-js/blob/6e0e02ce270fdac5731d7d59f1644aeb2dc081f6/src/GoTrueApi.ts#L281), so let us know if you're seeing any problems.

> How do you recommend handling the fact that the token will still persist and be valid? 

Welcome to JWT auth ... by default the JWT is only valid for 1 hour before it becomes invalid. You can revoke the token by deleting it on the user's side, but if they "stored" it somewhere technically they can keep using it for any hour. You can see that the library refreshes the token ever 60 minutes [here](https://github.com/supabase/gotrue-js/blob/6e0e02ce270fdac5731d7d59f1644aeb2dc081f6/src/GoTrueClient.ts#L396). 

Perhaps one day we will move to session-based auth, but at the moment the world prefers JWT auth :(

**Comment by gaurangrshah** [comment]
Once again I really appreciate you taking out the time to explain. Thank you very much. I'm gonna take a look thru and see what I can determine.
